System.IO.Abstractions.TestingHelpers/MockPath.cs

```diff
@@ -128,13 +132,18 @@ private string[] GetSegments(params string[] paths)
         public override string GetTempFileName()
         {
             string fileName = mockFileDataAccessor.Path.GetRandomFileName();
-            string tempDir = mockFileDataAccessor.Path.GetTempPath();
+            string tempDir = this.GetTempPath();
 
             string fullPath = mockFileDataAccessor.Path.Combine(tempDir, fileName);
 
             mockFileDataAccessor.AddFile(fullPath, new MockFileData(string.Empty));
 
             return fullPath;
         }
+
+        public override string GetTempPath()
+        {
+            return !string.IsNullOrEmpty(defaultTempDirectory) ? defaultTempDirectory : base.GetTempPath();
```

Comment:
Maybe move this logic to the constructor? Seems like making this decision once should be enough 😁